lnclient/lnd/lnd.go

```diff
@@ -448,14 +415,89 @@ func NewLNDService(ctx context.Context, lndAddress, lndCertHex, lndMacaroonHex s
 		return nil, err
 	}
 
-	lndService := &LNDService{client: lndClient, pubkey: info.IdentityPubkey}
+	lndCtx, cancel := context.WithCancel(ctx)
+
+	lndService := &LNDService{client: lndClient, pubkey: info.IdentityPubkey, cancel: cancel}
+
+	// Subscribe to payments
+	go func() {
+		paymentStream, err := lndClient.SubscribePayments(lndCtx, &routerrpc.TrackPaymentsRequest{
+			NoInflightUpdates: true,
+		})
+		if err != nil {
+			logger.Logger.Errorf("Error subscribing to payments: %v", err)
```

Comment:
instead of just returning (killing the gorouting completely) can we try again?

Reply:
You mean wrap it in a for loop and keep trying if we result in an error?

Reply:
Yeah, otherwise the subscription will stop working. It's similar to what we do when we reconnect to the relay

Reply:
Done!